tests/cxx/namespace_use.cc

```diff
@@ -0,0 +1,45 @@
+//===--- namespace_use.cc - test input file for iwyu ----------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// IWYU_ARGS: -I .
+
+// Tests that using-directives i.e. `using namespace`, get appropriate
+// include suggestions.
+
+#include "tests/cxx/namespace_use-i1.h"
```

Comment:
OK, so I think you've misunderstood the `-d` and `-i` convention (it's weird, so I don't blame you :)) --

* `*-dX.h` or `*-direct.h` indicate a _direct include_, a header the main file includes directly
* `*-iX.h` or `*-indirect.h` indicate an _indirect include_, a header that is indirectly included by a direct include

So the confusing thing is that direct includes are just necessary padding, they usually shouldn't contain anything -- their typical role is to be suggested for removal.

The indirect includes, on the other hand, are where the interesting symbols should go.

As far as I can tell, you've interpreted it exactly the other way around, so you should be able to "just" switch everything around.

Reply:
Whoops - I didn't think too hard when naming these, I just remembered there were d's and i's. Fixing.

Reply:
Thanks, makes more sense now!